src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -76,34 +76,59 @@ public int IndexOf(char value, int startIndex)
         }
 
         public int IndexOf(char value, StringComparison comparisonType)
+        {
+            return IndexOf(value, 0, comparisonType);
+        }
+
+        public int IndexOf(char value, int startIndex, StringComparison comparisonType)
+        {
+            return IndexOf(value, startIndex, Length - startIndex, comparisonType);
+        }
+
+        public int IndexOf(char value, int startIndex, int count, StringComparison comparisonType)
         {
             return comparisonType switch
             {
-                StringComparison.CurrentCulture or StringComparison.CurrentCultureIgnoreCase => CultureInfo.CurrentCulture.CompareInfo.IndexOf(this, value, GetCaseCompareOfComparisonCulture(comparisonType)),
-                StringComparison.InvariantCulture or StringComparison.InvariantCultureIgnoreCase => CompareInfo.Invariant.IndexOf(this, value, GetCaseCompareOfComparisonCulture(comparisonType)),
-                StringComparison.Ordinal => IndexOf(value),
-                StringComparison.OrdinalIgnoreCase => IndexOfCharOrdinalIgnoreCase(value),
+                StringComparison.CurrentCulture or StringComparison.CurrentCultureIgnoreCase => CultureInfo.CurrentCulture.CompareInfo.IndexOf(this, value, startIndex, count, GetCaseCompareOfComparisonCulture(comparisonType)),
+                StringComparison.InvariantCulture or StringComparison.InvariantCultureIgnoreCase => CompareInfo.Invariant.IndexOf(this, value, startIndex, count, GetCaseCompareOfComparisonCulture(comparisonType)),
+                StringComparison.Ordinal => IndexOf(value, startIndex, count),
+                StringComparison.OrdinalIgnoreCase => IndexOfCharOrdinalIgnoreCase(value, startIndex, count),
                 _ => throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType)),
             };
         }
 
-        private int IndexOfCharOrdinalIgnoreCase(char value)
+        private int IndexOfCharOrdinalIgnoreCase(char value, int startIndex, int count)
         {
-            if (!char.IsAscii(value))
+            ArgumentOutOfRangeException.ThrowIfNegative(startIndex);
+            ArgumentOutOfRangeException.ThrowIfGreaterThan(startIndex, Length);
+            ArgumentOutOfRangeException.ThrowIfNegative(count);
```

Comment:
Maybe better duplicate these two checks in the calling methods taking startIndex to make the empty string case work nicely.